### PR TITLE
Replay handler

### PR DESCRIPTION
**Pull request by aspacca**
Enhancement
## What does this PR do?
Introduces the replay handling mechanism

## Why is it important?
See https://github.com/elastic/elastic-serverless-forwarder/issues/54

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
~- [] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.md`

## Author's Checklist

- [ ]

## Related issues

-
Relates #54

### CODE REVIEW

**Comment by aspacca** [comment]
@v1v could you help to troubleshoot the failure?

I added an integration test in the same test class, that has a setup method where an elastic container is brought up.
since I added the second test, it is always failing due to connection timeout on the first es client operation in the test (while the operations in the setup -ping/health- not). like if the container died in the meanwhile or es took more than 10sec to send the response body

https://beats-ci.elastic.co/job/Library/job/elastic-serverless-forwarder-mbp/job/PR-60/9/testReport/tests.handlers.aws.test_handler/TestLambdaHandlerSuccess/test_lambda_handler_replay/

I cannot reproduce the error locally even running the tests on docker

**Comment by v1v** [comment]
> @v1v could you help to troubleshoot the failure?

Will it work if you can access to the CI worker and debug what's going on? If so, do you have access to the CI Workers through SSH? If you have access, I can lock one of the CI workers and provide you the hostname to connect to

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+### v0.17.0 - 2021/12/30
+##### Features
+* Replay queue for ES ingestion phase failure: [#60](https://github.com/elastic/elastic-serverless-forwarder/pull/60)
+
+
```
Comment (kaiyan-sheng) [inline CHANGELOG.md:5]:
```suggestion
```
one extra line here?

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**